certbot/cert_manager.py

```diff
@@ -20,3 +30,83 @@ def update_live_symlinks(config):
         storage.RenewableCert(renewal_file,
             configuration.RenewerConfiguration(renewer_config),
             update_symlinks=True)
+
+def _report_lines(msgs):
+    """Format a results report for a category of single-line renewal outcomes"""
+    certinfo = [str(msg) for msg in msgs]
+    return "  " + "\n  ".join(certinfo)
+
+def _report_human_readable(parsed_certs):
+    """Format a results report for a parsed cert"""
+    certinfo = []
+    for cert in parsed_certs:
+        now = pytz.UTC.fromutc(datetime.datetime.utcnow())
+        if cert.target_expiry <= now:
+            expiration_text = "EXPIRED"
+        else:
+            diff = cert.target_expiry - now
+            if diff.days == 1:
+                expiration_text = "1 day"
+            elif diff.days < 1:
+                expiration_text = "under 1 day"
+            else:
+                expiration_text = "{0} days".format(diff.days)
+        valid_string = "{0} ({1})".format(cert.target_expiry, expiration_text)
+        out = "  Lineage: {0}\n    Domains: {1}\n    Valid Until: {2}".format(
+            cert.lineagename, " ".join(cert.names()), valid_string)
+        certinfo.append(out)
+    return "\n".join(certinfo)
+
+def _describe_certs(config, parsed_certs, parse_failures):
+    """Print information about the certs we know about"""
+    out = []
+
+    notify = out.append
+
+    if not parsed_certs and not parse_failures:
+        notify("No certs found.")
+    else:
+        if parsed_certs:
+            notify("Found the following certs:")
+            notify(_report_human_readable(parsed_certs))
+        if parse_failures:
+            notify("\nThe following renewal configuration files "
+               "were invalid:")
+            notify(_report_lines(parse_failures))
+
+    if config.quiet:
+        return
```

Comment:
Yes that includes `IDisplay`. If `-q/--quiet` is provided, `zope.component.getUtility(interfaces.IDisplay)` returns a `NoninteractiveIDisplay` constructed [here](https://github.com/certbot/certbot/blob/master/certbot/main.py#L761).

I think we do this in `renewal.py` because we're using `print` not `IDisplay`.